components/script/dom/bluetooth.rs

```diff
@@ -40,12 +56,155 @@ impl Bluetooth {
+        let mut services: Vec<String> = vec!();
+        if filter.services.is_some() {
+            if filter.services.clone().unwrap().is_empty() {
+                return Err(Type("'services', if present, must contain at least one service.".to_owned()));
+            }
+            for service in filter.services.clone().unwrap() {
+                let validated_service = BluetoothUUID::GetService(global, service);
+                match validated_service {
+                    Err(err) => return Err(err),
+                    Ok(valid) => services.push(valid.to_string()),
+                }
+            }
+        }
+
+        let mut name = String::new();
+        if filter.name.is_some() {
+            // DOMString::len() gives back the size in bytes
+            if filter.name.clone().unwrap().len() > MAX_DEVICE_NAME_LENGTH {
+                return Err(Type(NAME_TOO_LONG_EXCEPTION.to_owned()));
+            }
+            if filter.name.clone().unwrap().len() > MAX_FILTER_NAME_LENGTH {
+                return Err(Type(FILTER_NAME_TOO_LONG_EXCEPTION.to_owned()));
+            }
+            name = filter.name.unwrap().to_string();
+        }
+
+        let mut name_prefix = String::new();
+        if filter.namePrefix.is_some() {
+            if filter.namePrefix.clone().unwrap().len() > MAX_DEVICE_NAME_LENGTH {
+                return Err(Type(NAME_TOO_LONG_EXCEPTION.to_owned()));
+            }
+            if filter.namePrefix.clone().unwrap().len() > MAX_FILTER_NAME_LENGTH {
+                return Err(Type(FILTER_NAME_TOO_LONG_EXCEPTION.to_owned()));
+            }
+            if filter.namePrefix.clone().unwrap().len() == 0 {
+                return Err(Type("'namePrefix', if present, must me non-empty.".to_owned()));
+            }
+            name_prefix = filter.namePrefix.clone().unwrap().to_string();
+        }
+
+        Ok(BluetoothScanfilter::new(name, name_prefix, services))
+}
+
+fn convert_request_device_options(options: &RequestDeviceOptions, global: GlobalRef)
+    -> Fallible<RequestDeviceoptions> {
+        // TODO: uncomment this after servo/master changes have been merged(zakorgy)
+        /*if options.filters.is_empty() {
+            return Err(Type("'filters' member must be non-empty to find any devices.".to_owned()));
+        }*/
+
+        let options = options.clone();
+        // TODO: replace the code above for this after servo/master changes merged(zakorgy)
+        /*
+        let mut filters = BluetoothScanfilterSequence::new(vec!());
+        for filter in options.filters {
+            let result = canonicalize_filter(filter, global);
+            match result {
+                Err(err) => return Err(err),
+                Ok(canonicalized_filter) => filters.push(canonicalized_filter);
+            }
+        }
+        */
+        // TODO: remove this code part after servo/master changes merged(zakorgy)
+        let result = canonicalize_filter(&options.filter, global);
```

Comment:
result is a good name; you might also see vars with the `_maybe` suffix for such values.